read-fonts/src/tables/hvar.rs

```diff
@@ -0,0 +1,98 @@
+//! The [HVAR (Horizontal Metrics Variation)](https://docs.microsoft.com/en-us/typography/opentype/spec/hvar) table
+
+use crate::variations::{DeltaSetIndex, DeltaSetIndexMap, ItemVariationStore};
+use font_types::Tag;
+
+/// 'HVAR'
+pub const TAG: Tag = Tag::new(b"HVAR");
+
+include!("../../generated/generated_hvar.rs");
+
+impl<'a> Hvar<'a> {
+    /// Returns the advance width delta for the specified glyph identifier and
+    /// normalized variation coordinates.
+    pub fn advance_width_delta(
+        &self,
+        glyph_id: GlyphId,
+        coords: &[F2Dot14],
+    ) -> Result<Fixed, ReadError> {
+        let gid = glyph_id.to_u16();
+        let ix = match self.advance_width_mapping() {
+            Some(Ok(dsim)) => dsim.get(gid as u32)?,
+            _ => DeltaSetIndex {
+                outer: 0,
+                inner: gid,
+            },
+        };
+        let ivs = self.item_variation_store()?;
+        ivs.compute_delta(ix, coords)
+    }
```

Comment:
Hvar::advance_width_delta and Vvar::advance_height_delta differ *only* in what ix is set to and which varstore is used. I feel a common function in your future.

Reply:
Good catch. I'll pull it out. Same with the side bearing/vorg code.